services/ownerService.js

```diff
@@ -43,13 +43,18 @@ exports.addHotel = async (data) => {
   return hotel
 }
 
-exports.updateHotel = async (id, data) => {
-  const hotelUpdate = await Hotel.findByIdAndUpdate(id, data)
+exports.updateHotel = async (id, data, userId) => {
+  let hotel = await Hotel.findById(id)
```

Comment:
Check something like findOneAndUpdate. You should be able to check if hotel exists and belongs to the hotel owner.